### PR TITLE
Feture/modal deploy

### PR DESCRIPTION
**Pull request by JackHamish**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
+  "rules": { "react/no-unescaped-entities": 0 },
```
Comment (anastasiia-tarasenko) [inline .eslintrc.json:2]: Why did you remove this rule?

File: src/app/nft/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    await deleteNftAction(nft.id, {
+      onSuccess() {
+        queryClient.invalidateQueries({ queryKey: ["nfts-user"] });
+        toast.success(`Nft ${nft.title} was deleted successful`);
+        router.back();
+      },
+    });
+  };
+
+  return (
```
Comment (anastasiia-tarasenko) [inline src/app/nft/[id]/page.tsx:30]:
```tsx
if (!nft) {
  return;
}

return <>...</>
```

File: src/app/nft/[id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                <div className="flex gap-3">
+                  <Image
+                    src="/images/avatar.png"
+                    alt="User Avatar"
+                    className="object-contain"
+                    width={24}
+                    height={24}
+                  />
+                  <span className="font-sans text-xl font-normal">
+                    {nft?.user.name}
```
Comment (anastasiia-tarasenko) [inline src/app/nft/[id]/page.tsx]: `nft` can't be undefined because you checked it above. So you don't need `?`

File: src/app/nft/[id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                </div>
+              </div>
+
+              <div className="flex flex-col gap-3">
+                <h4 className="text-2xl font-bold text-friar-gray">
+                  Description
+                </h4>
+
+                <p className="font-sans text-xl font-normal">
+                  {nft?.description}
```
Comment (anastasiia-tarasenko) [inline src/app/nft/[id]/page.tsx]:
```tsx
                  {nft.description}
```

File: src/app/profile/@createNftModal/loading.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import Modal from "@/components/modal";
+
+const Loading = () => {
+  return (
+    <Modal>
+      <p className="animate-bounce text-xl text-white">Loading ...</p>
```
Comment (anastasiia-tarasenko) [inline src/app/profile/@createNftModal/loading.tsx]: Move this line to a separate component because you copy this row of code many times.

File: src/app/profile/error.tsx
Comment (anastasiia-tarasenko) [inline src/app/profile/error.tsx:1]: If you have the same error handling on different pages, you can either use the error boundary in root or create a reusable component and reexport it in your error files.

File: src/components/categories/categories-card.tsx [hunk 6]
```diff
@@ -1,34 +1,27 @@
...
-                        width={100}
-                        height={100}
-                    />
-                </div>
-            </div>
-            <h3 className="py-5 px-7 font-sans text-xl font-semibold">
-                {title}
-            </h3>
+  return (
+    <div className="mt-10 w-60 rounded-3xl bg-mine-shaft-darken">
```
Comment (anastasiia-tarasenko) [inline src/components/categories/categories-card.tsx]: Remove 'mt-10', you used gap for cards.

File: src/components/creators/creator-card.tsx [hunk 7]
```diff
@@ -1,38 +1,34 @@
...
-                </h3>
-                <div className="mt-2 text-center flex w-full">
-                    <h4 className="text-base font-sans text-friar-gray">
-                        Total Sales:
-                    </h4>
-                    <span className="ml-3 text-base">{totalSales} ETH</span>
-                </div>
-            </div>
+  return (
+    <div className="relative mt-10 flex w-60 flex-col items-center justify-center rounded-3xl bg-mine-shaft-darken p-5">
```
Comment (anastasiia-tarasenko) [inline src/components/creators/creator-card.tsx]: Remove `mt-10`

**Review by anastasiia-tarasenko — APPROVED** [review]
Awesome work!

File: src/utils/firebase/firebase.ts [hunk 8]
```diff
@@ -12,5 +12,3 @@ const firebaseConfig = {
 };
 
 export const firebaseApp = initializeApp(firebaseConfig);
-export const messaging = getMessaging(firebaseApp);
```
Comment (lluchkaa) [inline src/utils/firebase/firebase.ts:15]: actually, it was a good idea to keep it here

File: src/domains/nft/types.ts [hunk 9]
```diff
@@ -1,17 +1,21 @@
 import { User } from "@/types/user";
 
-export type NftData = {
-  id: string;
+export type NftFullData = NftSmallData & {
```
Comment (lluchkaa) [inline src/domains/nft/types.ts]: I usually name it like this:
```ts
type Product = ...
type ProductDetails = Product & ...

File: src/domains/nft/hooks.ts [hunk 10]
```diff
@@ -8,6 +8,12 @@ export const useUserNfts = () =>
     queryFn: getAllUserNfts,
   });
 
+export const useCurrentNft = (id: string) =>
+  useQuery({
+    queryKey: ["nfts-current"],
+    queryFn: () => getNftById(id),
+  });
+
```
Comment (lluchkaa) [inline src/domains/nft/hooks.ts:16]: Is there a reason to call it current nft?
Could be like this:
```suggestion
export const useNft = (id: string) =>
  useQuery({
    queryKey: ["nfts", id],
    queryFn: () => getNftById(id),
  });

```

File: src/components/providers/notifications-provider.tsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+      const unsubscribe = onMessage(messaging, (payload) => {
+        toast.info(payload.notification?.body);
+      });
+      return () => {
+        unsubscribe();
+      };
+    }
+  }, []);
+
+  return <></>;
```
Comment (lluchkaa) [inline src/components/providers/notifications-provider.tsx]:
```suggestion
  return null;
```

File: src/components/providers/notifications-provider.tsx [hunk 12]
```diff
@@ -0,0 +1,26 @@
+"use client";
+
+import { firebaseApp } from "@/utils/firebase/firebase";
+import useFcmToken from "@/utils/firebase/useFcmToken";
+import { getMessaging, onMessage } from "firebase/messaging";
+import { useEffect } from "react";
+import { toast } from "react-toastify";
+
+export const NotificationsProvider = () => {
```
Comment (lluchkaa) [inline src/components/providers/notifications-provider.tsx]: It feels more like a `Listener`, not a `Provider`

File: src/app/profile/@nftModal/default.tsx
Comment (lluchkaa) [inline src/app/profile/@nftModal/default.tsx:1]: Looks like you can remove this and other empty `default.tsx` files

File: src/app/profile/@createNftModal/(.)create-nft/form/constants.ts [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import z from "zod";
+
+export const MAX_DESCRIPTION_HEIGHT = 192;
+
+export const createNftSchema = z.object({
+  title: z.string(),
+  description: z.string().max(179, { message: "Description is too long" }),
+  file: z.instanceof(File, { message: "Bad file" }),
```
Comment (lluchkaa) [inline src/app/profile/@createNftModal/(.)create-nft/form/constants.ts:8]: Would be great to add file size validation

File: src/app/nft/[id]/page.tsx [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+  const queryClient = useQueryClient();
+
+  const { mutateAsync: deleteNftAction } = useDeleteNft();
+
+  const handleClickDelete = async () => {
+    if (!nft) return;
+
+    await deleteNftAction(nft.id, {
+      onSuccess() {
+        queryClient.invalidateQueries({ queryKey: ["nfts-user"] });
```
Comment (lluchkaa) [inline src/app/nft/[id]/page.tsx]: This should be done inside the `useDeleteNft` hook.

File: src/app/profile/@createNftModal/(.)create-nft/form/index.tsx [hunk 15]
```diff
@@ -1,37 +1,60 @@
...
+      return;
+    }
+
+    descriptionInput.style.height = "auto";
+    descriptionInput.style.height = `${
+      descriptionInput.scrollHeight < MAX_DESCRIPTION_HEIGHT
+        ? descriptionInput.scrollHeight
+        : MAX_DESCRIPTION_HEIGHT
+    }px`;
+  };
```
Comment (lluchkaa) [inline src/app/profile/@createNftModal/(.)create-nft/form/index.tsx:41]: It's better to encapsulate this inside some `AutoHeightTextArea` component

File: src/app/profile/@createNftModal/(.)create-nft/form/index.tsx [hunk 16]
```diff
@@ -40,10 +63,13 @@ const CreateNftForm = () => {
     try {
       await createNFtAction(data, {
         onSuccess(data, variables, context) {
+          queryClient.invalidateQueries({ queryKey: ["nfts-user"] });
```
Comment (lluchkaa) [inline src/app/profile/@createNftModal/(.)create-nft/form/index.tsx]: Move this inside the `useAddNft`

**Review by lluchkaa — APPROVED** [review]
Greaet job!
